internal/validation/validation.go

```diff
@@ -0,0 +1,115 @@
+// Package validation implements functions that validate requests data.
+package validation
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+)
+
+const (
+	// emailRegex checks the validity of the format of the entire email address.
+	emailRegex = "^[A-Za-z0-9._]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$"
+
+	// passwordOneLowercaseRegex searches for at least one lowercase character in password.
+	passwordOneLowercaseRegex = "(.*[a-z])"
+
+	// passwordOneUppercaseRegex searches for at least one uppercase character in password.
+	passwordOneUppercaseRegex = "(.*[A-Z])"
+
+	// passwordOneDigitRegex searches for at least one digit in password.
+	passwordOneDigitRegex = "(.*[0-9])"
+
+	// filenameInvalidCharactersRegex searches for special characters in password.
+	filenameInvalidCharactersRegex = "[:#%^;<>\\{}[\\]+~`=?&,\" ]"
+)
+
+const (
+	minPasswordLength = 8
+	maxPasswordLength = 20
+	minEmailLength    = 8
+	minRatio          = 1
+	maxRatio          = 9
+)
+
+var formats = map[string]struct{}{
+	"jpeg": {},
+	"png":  {},
+}
+
+var (
+	errInvalidEmailLength = fmt.Errorf("invalid email: email minimum %d characters", minPasswordLength)
+	errInvalidEmailFormat = errors.New("invalid email address format")
+
+	errInvalidPasswordLength    = fmt.Errorf("invalid password: length must be from %d to %d characters", minPasswordLength, maxPasswordLength)
+	errNoOneLowercaseInPassword = errors.New("invalid password: must contain at least one lowercase character")
+	errNoOneUppercaseInPassword = errors.New("the password: must contain at least one uppercase character")
+	errNoOneDigitInPassword     = errors.New("the password: must contain at least one digit")
+
+	errMissingFilename       = errors.New("missing filename")
+	errInvalidFilenameFormat = errors.New("invalid filename: shouldn't contain space and any special characters like :;<>{}[]+=?&,\"")
+
+	errInvalidSourceFormat = errors.New("invalid source format: needed jpeg, jpg or png")
+	errInvalidTargetFormat = errors.New("invalid source format: needed jpeg, jpg or png")
+	errEqualsFormats       = errors.New("source and target formats should differ")
+
+	errInvalidRatio = fmt.Errorf("invalid ratio: needed a value from %d to %d inclusive", minRatio, maxRatio)
+)
```

Comment:
These are all the same type of error "invalid parameter", it's the parameter itself is different